lib/Sema/CSDiag.cpp

```diff
@@ -2430,6 +2430,12 @@ bool FailureDiagnosis::diagnoseGeneralMemberFailure(Constraint *constraint) {
     }
   }
 
+  // If base type has unresolved generic parameters, such might mean
+  // that it's initializer with erroneous argument, otherwise this would
+  // be a simple ambigious archetype case, neither can be diagnosed here.
+  if (baseTy->hasTypeParameter())
```

Comment:
I don't think so because while type-checking anchor of the expression, it's allowed to produce unresolved types, so in case of `Array(.n).init()` it produced `Array<<<unresolved>>.Iterator.Element>`